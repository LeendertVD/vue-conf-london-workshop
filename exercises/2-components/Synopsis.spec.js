import Synopsis from './Synopsis.vue'
import { shallowMount } from '@vue/test-utils'

// You can use the Vue Test Utils slots option to pass in a slot:
//
// const wrapper = shallowMount(Component, {
//   slots: {
//     foo: '<div />',
//     bar: '<p>bar</p>',
//   }
// })
//
// https://vue-test-utils.vuejs.org/api/options.html#slots
test('renders short slot, and hides long slot, initially', () => {
    // Arrange
    const short = 'Short synopsis'
    const long = 'Long synopsis'
    const wrapper = shallowMount(Synopsis, {
        slots: {
        short: `<p>${short}</p>`,
        long: `<p>${long}</p>`
        }
    })

    // Assert
    expect(wrapper.text()).toContain(short)
    expect(wrapper.text()).not.toContain(long)
})

// You can use the Vue Test Utils find and trigger methods to trigger an event:
//
// const wrapper = shallowMount(Component)
// wrapper.find('button').trigger('click')
//
// https://vue-test-utils.vuejs.org/api/options.html#slots
test('renders long slot, and hides short slot, when button is clicked', () => {
    // Arrange
    const short = 'Short synopsis'
    const long = 'Long synopsis'
    const wrapper = shallowMount(Synopsis, {
      slots: {
        short: `<p>${short}</p>`,
        long: `<p>${long}</p>`
      }
    })

    wrapper.find('button').trigger('click')

    // Assert
    expect(wrapper.text()).not.toContain(short)
    expect(wrapper.text()).toContain(long)
})

test('toggles "Show more/ show less" when button is clicked', () => {
    // Arrange
    const short = 'Short synopsis'
    const long = 'Long synopsis'
    const wrapper = shallowMount(Synopsis, {
        slots: {
        short,
        long
        }
    })
    
    // Assert
    expect(wrapper.find('button').text()).toContain('Show more')

    // Act
    wrapper.find('button').trigger('click')

    // Assert
    expect(wrapper.find('button').text()).toContain('Show less')
    
})

test('toggles "Show more/ show less" when button is clicked', () => {
    // Arrange
    const short = 'Short synopsis'
    const long = 'Long synopsis'
    const wrapper = shallowMount(Synopsis, {
        slots: {
        short,
        long
        }
    })
    
    // Assert
    expect(wrapper.find('button').text()).toContain('Show more')

    // Act
    wrapper.find('button').trigger('click')

    // Act
    expect(wrapper.element).toMatchSnapshot();
})
