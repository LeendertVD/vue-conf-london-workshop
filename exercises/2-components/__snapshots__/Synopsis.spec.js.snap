// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toggles "Show more/ show less" when button is clicked 1`] = `
<div>
  <!---->
    
  <button>
    
    Show less
  
  </button>
</div>
`;
